addons/mail/static/src/web/chatter.xml

```diff
@@ -156,7 +156,7 @@
     <button class="btn btn-link text-action px-1 d-flex align-items-center" aria-label="Attach files" t-att-class="{ 'my-2': !props.compactHeight }" t-on-click="onClickAddAttachments">
         <i class="fa fa-paperclip fa-lg me-1"/>
         <span t-if="attachments.length > 0" t-esc="attachments.length"/>
-        <i t-if="!state.thread.areAttachmentsLoaded and state.thread.isLoadingAttachments" class="fa fa-circle-o-notch fa-spin" aria-label="Attachment counter loading..."/>
+        <i t-if="!state.thread.areAttachmentsLoaded and state.thread.isLoadingAttachments and state.showAttachmentLoading" class="fa fa-circle-o-notch fa-spin" aria-label="Attachment counter loading..."/>
```

Comment:
I kept `state.thread.isLoadingAttachments` otherwise test "Attachments that have been unlinked from server should be visually unlinked from record" was failing at last step.

The counter was not updated accordingly, still `2` instead of `1`, due to `state.thread.isLoadingAttachments` not being observed as needed somehow.